alibi_detect/cd/base.py

```diff
@@ -732,38 +660,12 @@ def predict(self, x: Union[np.ndarray, list], return_p_val: bool = True, return_
             cd['data']['distance_threshold'] = distance_threshold
         return cd
 
-    @abstractmethod
-    def get_config(self) -> dict:
-        """
-        Get the detector's configuration dictionary.
 
-        Returns
-        -------
-        The detector's configuration dictionary.
-        """
-        cfg = self.drift_config()
-
-        # Detector field
-        kwargs = {
-            'p_val': self.p_val,
-            'x_ref_preprocessed': self.preprocess_at_init or self.x_ref_preprocessed,
-            'preprocess_at_init': self.preprocess_at_init,
-            'update_x_ref': self.update_x_ref,
-            'configure_kernel_from_x_ref': not self.infer_sigma,
-            'n_permutations': self.n_permutations,
-            'input_shape': self.input_shape,
-            'data_type': self.meta['data_type'],
-            'backend': self.meta['backend'],
-        }
-        cfg.update(kwargs)
-
-        return cfg
-
-
-class BaseLSDDDrift(BaseDetector, DriftConfigMixin):
+class BaseLSDDDrift(BaseDetector):
     # TODO: TBD: this is only created when _configure_normalization is called from backend-specific classes,
     # is declaring it here the right thing to do?
     _normalize: Callable
+    _unnormalize: Callable
```

Comment:
Why do we need the inverse transform?

Reply:
The `LSDDDrift` preprocess logic is a little different to the other detectors since `self.x_ref` is normalized (regardless of whether `x_ref` is preprocessed by `preprocess_fn`):

https://github.com/SeldonIO/alibi-detect/blob/644ec298f73674175e605696ceeab5859864198a/alibi_detect/cd/tensorflow/lsdd.py#L83-L87

To avoid having to add another `x_ref_preprocessed` kwarg specific to whether `x_ref` has been normalized, the easiest thing to do is just unormalise `x_ref` before returning the config:

https://github.com/SeldonIO/alibi-detect/blob/644ec298f73674175e605696ceeab5859864198a/alibi_detect/cd/lsdd.py#L151-L156

Reply:
I see, unrelated but to be honest I find the following condition very confusing:
```python
if self.preprocess_at_init or self.preprocess_fn is None or self.x_ref_preprocessed
```

So the data is normalized when either:
 - we specify that we want to pre-process at init (but then preprocessing function doesn't have to exist?)
 - we don't have a preprocessing function of our own (??)
 - the data is already preprocessed (which implies a preprocessing function was passed?)

It seems like there is something missing here...

Reply:
I see it's explained below, will dig into that...